app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_results = {}
+
+    def inner(*args: Any) -> Any:
+        if args not in cache_results:
+            print("Calculating new result")
+            result = func(*args)
+            cache_results[args] = result
+        else:
+            print("Getting from cache")
+            result = cache_results[args]
+        return result
```

Comment:
You can return here just cache_results[args] , instead of writing this value to a variable 'result' Besides which you already have on 10 line , try always use different variables naming